File: 1022. Sum of Root To Leaf Binary Numbers/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumRootToLeaf(self, root: TreeNode,val=0) -> int:
        def dfs(node, prefix):
            if node:
                new_prefix = node.val + (prefix << 1)

                if node.left or node.right:
                    return dfs(node.left, new_prefix) + dfs(node.right, new_prefix)
                else:
                    return new_prefix
            else:
                return 0

        return dfs(root, 0)
                
        

File: 1022. Sum of Root To Leaf Binary Numbers/main.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int ans = 0;
    public int sumRootToLeaf(TreeNode root) {
        dfs(root,0);
        return ans;
    }
    private void dfs(TreeNode root, int val){
        if (root==null) return;
        val = val << 1 | root.val;
        if(root.left == null && root.right == null) ans+= val;
        dfs(root.left,val);
        dfs(root.right,val);
    }
}